Bugfix

qc-azarrabi opened the pull request:
Multiple bugfixes and updates.

qc-azarrabi committed:
libqcomtee: fix ioctl arguments

Add missing '&' for buf_data.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: fix sanity check conditions on multiple functions

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: set archive as default artifact

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

quicbasantk commented on libqcomtee/CMakeLists.txt:
```diff
@@ -28,7 +28,7 @@ set(SRC
 )
 
-add_library(qcomtee SHARED ${SRC})
+add_library(qcomtee ${SRC})
```
Why are we making static library?

qc-azarrabi replied:
It makes is easier for the testing; we can later convert to config or both.

qc-azarrabi committed:
test: kill the server thread

On IOCTL failure to wait, there is no reason to loop.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: fix number of parameter for send buffer

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: reset memory

Driver expects it.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: some doc cleanup

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: add support for kernel's UBUF output size

Kernel will return the size of expected output buffer, update
to support that.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: update credentials object

Update based on commit ab70814e9f19

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>
